File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {
    public int[] solution(int k, int[] score) {
        int[] answer = new int[score.length];
        List<Integer> list = new ArrayList<>();

        for (int i = 0; i < score.length; i++) {
            list.add(score[i]);
            list.sort(Collections.reverseOrder());

            if (list.size() < k) {
                answer[i] = list.get(list.size() - 1);
            } else {
                answer[i] = list.get(k - 1);
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    rank = []
    
    for i in score:
        rank.append(i)
        rank = sorted(rank, reverse = True)
        
        if len(answer) < k:
            answer.append(rank[-1])
        else:
            answer.append(rank[k - 1])
            
    return answer